indexer/feature_meta.cpp

```diff
@@ -50,11 +50,24 @@ std::string Metadata::GetWikiURL() const
   return ToWikiURL(string(Get(FMD_WIKIPEDIA)));
 }
 
-string Metadata::ToWikimediaCommonsURL(std::string const & v)
+std::string Metadata::ToWikimediaCommonsURL(std::string v)
 {
   if (v.empty())
     return v;
 
+  // Spaces and ? characters should be corrected to form a valid URL's path.
+  for (auto i = 0; i < v.size(); ++i)
```

Comment:
If this code is similar/the same as `Metadata::ToWikiURL`, please move it into a reusable static function (or into the anonymous namespace), and call it from both these methods.

Reply:
I'm not following the logic.
1. Why percent escaping is not needed?
2. Why the same code should not be reused but instead copy-pasted?